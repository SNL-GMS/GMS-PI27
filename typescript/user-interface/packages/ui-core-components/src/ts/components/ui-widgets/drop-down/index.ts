export type { DropDownProps } from './drop-down';
export { DropDown } from './drop-down';
