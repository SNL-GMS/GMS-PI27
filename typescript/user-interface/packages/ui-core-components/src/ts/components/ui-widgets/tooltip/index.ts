import { HotkeyTooltip } from './hotkey-tooltip';
import { TooltipWrapper } from './tooltip';

export { HotkeyTooltip, TooltipWrapper as Tooltip2Wrapper };
