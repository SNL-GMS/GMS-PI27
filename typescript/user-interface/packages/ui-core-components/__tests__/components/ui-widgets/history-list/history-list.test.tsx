/* eslint-disable react/jsx-props-no-spreading */
import { fireEvent, render, screen } from '@testing-library/react';
import * as React from 'react';

import { HistoryList } from '../../../../src/ts/components/ui-widgets/history-list';
import type * as HistoryListTypes from '../../../../src/ts/components/ui-widgets/history-list/types';

const historyListItem: HistoryListTypes.HistoryListItem = {
  index: 0,
  label: 'item A',
  id: '1'
};

const historyListItem2: HistoryListTypes.HistoryListItem = {
  index: 1,
  label: 'item B',
  id: '2'
};

const props: HistoryListTypes.HistoryListProps = {
  items: [historyListItem],
  preferredItems: [historyListItem],
  listLength: 1,
  onSelect: jest.fn()
};

describe('History List', () => {
  it('to be defined', () => {
    expect(HistoryList).toBeDefined();
  });

  it('History List renders', () => {
    const { container } = render(<HistoryList {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('shows a defined number of items', () => {
    const { rerender } = render(
      <HistoryList items={[historyListItem]} listLength={1} onSelect={jest.fn()} />
    );

    expect(screen.getAllByRole('menuitem')).toHaveLength(1);

    rerender(
      <HistoryList
        items={[historyListItem, historyListItem2]}
        listLength={1}
        onSelect={jest.fn()}
      />
    );

    expect(screen.getAllByRole('menuitem')).toHaveLength(1);

    rerender(<HistoryList items={[historyListItem, historyListItem2]} onSelect={jest.fn()} />);

    expect(screen.getAllByRole('menuitem')).toHaveLength(2);
  });

  it('preferred items are not limited by the list length', () => {
    render(
      <HistoryList
        items={[historyListItem]}
        preferredItems={[historyListItem, historyListItem2]}
        listLength={1}
        onSelect={jest.fn()}
      />
    );

    expect(screen.getAllByRole('menuitem')).toHaveLength(3);
  });

  it('triggers a callback on click', () => {
    const onSelectMock = jest.fn();

    render(
      <HistoryList
        items={[historyListItem]}
        preferredItems={[historyListItem2]}
        listLength={1}
        onSelect={onSelectMock}
      />
    );
    fireEvent.click(screen.getByText(historyListItem.label));

    expect(onSelectMock).toHaveBeenCalledWith(historyListItem.id);

    fireEvent.click(screen.getByText(historyListItem2.label));

    expect(onSelectMock).toHaveBeenCalledWith(historyListItem2.id);
  });
});
