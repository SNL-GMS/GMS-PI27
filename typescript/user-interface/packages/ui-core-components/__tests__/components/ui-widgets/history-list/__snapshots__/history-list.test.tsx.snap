// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History List History List renders 1`] = `
<div>
  <div>
    <ul
      class="bp5-menu"
      role="menu"
    >
      <li
        class=""
        role="none"
      >
        <a
          class="bp5-menu-item bp5-popover-dismiss"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="bp5-text-overflow-ellipsis bp5-fill"
          >
            item A
          </div>
        </a>
      </li>
      <li
        class="bp5-menu-divider"
        role="separator"
      />
      <li
        class=""
        role="none"
      >
        <a
          class="bp5-menu-item bp5-popover-dismiss"
          role="menuitem"
          tabindex="0"
        >
          <div
            class="bp5-text-overflow-ellipsis bp5-fill"
          >
            item A
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
